src/sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal.ts

```diff
@@ -0,0 +1,326 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Checkbox } from 'sql/base/browser/ui/checkbox/checkbox';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { IClipboardService } from 'sql/platform/clipboard/common/clipboardService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { Modal } from 'sql/workbench/browser/modal/modal';
+import { attachModalDialogStyler } from 'sql/workbench/common/styler';
+import { ITextResourcePropertiesService } from 'vs/editor/common/services/textResourceConfigurationService';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { IColorTheme, ICssStyleCollector, IThemeService, registerThemingParticipant } from 'vs/platform/theme/common/themeService';
+import * as DOM from 'vs/base/browser/dom';
+import { attachCheckboxStyler } from 'sql/platform/theme/common/styler';
+import { ServiceOptionType } from 'sql/platform/connection/common/interfaces';
+import { ServiceOption } from 'azdata';
+import * as DialogHelper from 'sql/workbench/browser/modal/dialogHelper';
+import { TextCellComponent } from 'sql/workbench/contrib/notebook/browser/cellViews/textCell.component';
+import { ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
+import { ICellModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { NotebookModel } from 'sql/workbench/services/notebook/browser/models/notebookModel';
+import { inputBorder, inputValidationInfoBorder, editorBackground } from 'vs/platform/theme/common/colorRegistry';
+import { localize } from 'vs/nls';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { attachButtonStyler } from 'vs/platform/theme/common/styler';
+import { toJpeg } from 'html-to-image';
+
+export class CellOptionsModel {
+	private _optionsMap: { [name: string]: any } = {};
+
+	constructor(
+		optionsMetadata: ServiceOption[],
+		private onInsert: (cell: ICellModel) => any,
+		private _context: NotebookViewsExtension,
+	) {
+		optionsMetadata.forEach(optionMetadata => {
+			let defaultValue = this.getDisplayValue(optionMetadata, optionMetadata.defaultValue);
+			this._optionsMap[optionMetadata.name] = {
+				optionMetadata: optionMetadata,
+				defaultValue: defaultValue,
+				currentValue: defaultValue
+			};
+		});
+	}
+
+	private getDisplayValue(optionMetadata: ServiceOption, optionValue: any): any {
```

Comment:
It would be great to reduce usage of any here...so please add proper typings as possible

Reply:
Added a type for optionValue, and replaced the rest of the uses of any in this file.